src/Clock.tsx

```diff
@@ -0,0 +1,54 @@
+import { Component } from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  date: Date,
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      let { date } = this.state;
+
+      date = new Date();
+      this.setState({ date });
+      // eslint-disable-next-line no-console
+      console.log(date.toLocaleTimeString());
```

Comment:
Why do you do this? You don't use date from state, so let's just:
```suggestion
      const date = new Date();
      this.setState({ date });
      // eslint-disable-next-line no-console
      console.log(date.toLocaleTimeString());
```